addons/pinniped/config-controller/controllers/utils_test.go

```diff
@@ -131,3 +131,85 @@ func verifySecretFunc(ctx context.Context, cluster *clusterapiv1beta1.Cluster, c
 		}
 	}
 }
+
+// validates the expected create, update changes for a v1alpha1 pinniped secret
+func verifyV1alpha1SecretUpdateFunc(ctx context.Context, cluster *clusterapiv1beta1.Cluster, configMap *corev1.ConfigMap) func(Gomega) {
+	return func(g Gomega) {
+		clusterCopy := cluster.DeepCopy()
+		err := k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterCopy), clusterCopy)
+		g.Expect(err).NotTo(HaveOccurred())
+
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: clusterCopy.Namespace,
+				Name:      fmt.Sprintf("%s-pinniped.tanzu.vmware.com-package", clusterCopy.Name),
+			},
+		}
+
+		wantValuesYAML := makeSecretYAMLwithOIDC(cluster, configMap)
+
+		wantSecretLabel := map[string]string{}
+		wantSecretLabel[tkgAddonLabel] = pinnipedAddonLabel
+		secret.Name = fmt.Sprintf("%s-pinniped-addon", clusterCopy.Name)
+
+		err = k8sClient.Get(ctx, client.ObjectKeyFromObject(secret), secret)
+
+		g.Expect(secret.Labels).To(Equal(wantSecretLabel))
+		var gotValuesYAML map[string]interface{}
+		g.Expect(yaml.Unmarshal(secret.Data[tkgDataValueFieldName], &gotValuesYAML)).Should(Succeed())
+		g.Expect(gotValuesYAML).Should(Equal(wantValuesYAML))
+
+		g.Expect(string(secret.Data[tkgDataValueFieldName])).To(HavePrefix(valuesYAMLPrefix))
+		g.Expect(secret.Type).To(Equal(corev1.SecretType(tkgAddonType)))
+		g.Expect(secret.Annotations).To(Equal(map[string]string{tkgAddonTypeAnnotation: pinnipedAddonTypeAnnotation}))
+	}
+}
+
+// validates the expected delete changes for a v1alpha1 pinniped secret
+func verifyV1alpha1SecretDeleteFunc(ctx context.Context, cluster *clusterapiv1beta1.Cluster) func(Gomega) {
+	return func(g Gomega) {
+		clusterCopy := cluster.DeepCopy()
+		err := k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterCopy), clusterCopy)
+		g.Expect(err).NotTo(HaveOccurred())
+
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: clusterCopy.Namespace,
+				Name:      fmt.Sprintf("%s-pinniped.tanzu.vmware.com-package", clusterCopy.Name),
+			},
+		}
+		err = k8sClient.Get(ctx, client.ObjectKeyFromObject(secret), secret)
+		g.Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+	}
+}
+
+// TODO: maybe do this?
```

Comment:
If this seems valuable, I can split out the rest and fill out this func also.

Reply:
It seems like a lot of repeated code right now, but def down to do some refactoring later to resolve the issues you have.

Reply:
Removing it for now, but definitely a good conversation.  I worry that ALL tests run through this function, so its high risk to touch it, and we have to touch it every time we make updates.

I'm all for a little repeated [test] code for safety nets (unrelated tests that don't change when we make updates).